source/Body.h

```diff
@@ -99,48 +122,58 @@ class Body {
 	Point position;
 	Point velocity;
 	Angle angle;
-	Point center;
 	Point rotatedCenter;
 	// A zoom of 1 means the sprite should be drawn at half size. For objects
 	// whose sprites should be full size, use zoom = 2.
 	float zoom = 1.f;
-	float scale = 1.f;
+	mutable float scale = 1.f;
+	mutable bool debug = false;
```

Comment:
What is this for?

Reply:
Scale refers to the size of the sprite. Since the scale can be different across different sprites of the same ship (depending on design), I added it as a part of SpriteParameters. Every time a ship enters a new state, the SpriteParameters for that state is requested which causes the mutable variables to be refreshed according to the new SpriteParameters.

Debug was just something I had introduced to test an issue with the the trigger sprites and I forgot to remove it. I will remove it now.